src/main/java/appeng/core/worlddata/SpatialDimensionManager.java

```diff
@@ -0,0 +1,245 @@
+		{
+			return getBlockPosFromId( cellStorageId );
+		}
+		return null;
+	}
+
+	@Override
+	public BlockPos getCellContentSize( int cellStorageId )
+	{
+		StorageCellData cell = this.spatialData.get( cellStorageId );
+		if( cell != null )
+		{
+			return cell.contentDimension;
+		}
+		return null;
+	}
+
+	@Override
+	public boolean hasCapability( Capability<?> capability, EnumFacing facing )
+	{
+		return capability == Capabilities.SPATIAL_DIMENSION;
+	}
+
+	@Override
+	public <T> T getCapability( Capability<T> capability, EnumFacing facing )
+	{
+		if( capability == Capabilities.SPATIAL_DIMENSION )
+		{
+			return (T) this;
+		}
+		return null;
+	}
+
+	@Override
+	public NBTTagCompound serializeNBT()
+	{
+		final NBTTagCompound ret = new NBTTagCompound();
+		final NBTTagList list = new NBTTagList();
+
+		for( Map.Entry<Integer, StorageCellData> entry : this.spatialData.entrySet() )
+		{
+			final NBTTagCompound nbt = entry.getValue().serializeNBT();
+			nbt.setInteger( NBT_SPATIAL_ID_KEY, entry.getKey() );
+			list.appendTag( nbt );
+		}
+		ret.setTag( NBT_SPATIAL_DATA_KEY, list );
+		return ret;
+	}
+
+	@Override
+	public void deserializeNBT( NBTTagCompound nbt )
+	{
+		if( nbt.hasKey( NBT_SPATIAL_DATA_KEY ) )
+		{
+			final NBTTagList list = (NBTTagList) nbt.getTag( NBT_SPATIAL_DATA_KEY );
+
+			this.spatialData.clear();
+			for( int i = 0; i < list.tagCount(); ++i )
+			{
+				final NBTTagCompound entry = list.getCompoundTagAt( i );
+				final StorageCellData data = new StorageCellData();
+				final int id = entry.getInteger( NBT_SPATIAL_ID_KEY );
+				data.deserializeNBT( entry );
+				this.spatialData.put( id, data );
+			}
+		}
+	}
+
+	private int getNextId()
+	{
+		return this.spatialData.keySet().stream().max( Integer::compare ).orElse( -1 ) + 1;
+	}
+
+	private BlockPos getBlockPosFromId( int id )
+	{
+		int signBits = id & 0b11;
+		int offsetBits = id >> 2;
+		int offsetScale = 1;
+		int posx = MAX_CELL_DIMENSION / 2;
+		int posz = MAX_CELL_DIMENSION / 2;
+
+		// find quadrant
+		while( offsetBits != 0 )
+		{
+			posx += MAX_CELL_DIMENSION * offsetScale * ( offsetBits & 0b01 );
+			posz += MAX_CELL_DIMENSION * offsetScale * ( offsetBits >> 1 & 0b01 );
+
+			offsetBits >>= 2;
+			offsetScale <<= 1;
+		}
+
+		// mirror in one of 4 directions
+		if( ( signBits & 0b01 ) == 0 )
+		{
+			posx *= -1;
+		}
+		if( ( signBits & 0b10 ) == 0 )
+		{
+			posz *= -1;
+		}
+
+		// offset from cell center
+		posx -= 64;
+		posz -= 64;
+
+		return new BlockPos( posx, 64, posz );
+	}
+
+	private void clearCellArea( int cellId, StorageCellData cell )
+	{
+		// TODO reset chunks?
+	}
+
+	private static class StorageCellData implements INBTSerializable<NBTTagCompound>
+	{
+		private static final String NBT_OWNER_KEY = "owner";
+		private static final String NBT_DIM_X_KEY = "dim_x";
+		private static final String NBT_DIM_Y_KEY = "dim_y";
+		private static final String NBT_DIM_Z_KEY = "dim_z";
+
+		public BlockPos contentDimension;
+		public int owner;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make owner a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)